index.js

```diff
@@ -551,6 +552,19 @@ Browserify.prototype._createDeps = function (opts) {
                 return through();
             }
         }
+
+        if (opts.commondir === false && opts.builtins === false) {
+          opts.insertGlobalVars = xtend({
+            __dirname: function(file, basedir) {
+              var dir = path.dirname(path.relative(basedir, file));
+              return 'require("path").join(__dirname,"' + dir.split(path.sep).join('","') + '")';
+            },
+            __filename: function(file) {
+              var filename = path.relative(basedir, file);
```

Comment:
Yeah, I guess only a linter is able to automatically spot 'conflicting' variable names and suggest that one of them gets renamed, but that's obviously well outside the scope of this PR ;-)